### DISCUSSION TITLE
database - ref doesn't work

### DISCUSSION THREAD

**Discussion by AttilaBerczik** [invertase/react-native-firebase#5381] [Q&A]
Hi,
I can't read or write into my database, so I was trying to figure out the problem.

When I run `console.log(database().ref("/users/123"));`
I get `"null/users/123"`

But when I try this with firestore `console.log(firestore().doc("users/" + userId));`
I get a long message `{"_documentPath": {"_parts": ["users", "Gz7eG1y7qWX6Q9gijSPDlcLcZop1"]},` ...

When I run `console.log(database());`
I get a long message `{"_app": {"_automaticDataCollectionEnabled": true, "_deleteApp": [Function bound deleteApp],` ...

From this I assume that my firebase setup is correct, there should be a problem with the ref function.

Could you guys help?

I use `   "@react-native-firebase/app": "^12.0.0",
    "@react-native-firebase/database": "^12.0.0",
    "@react-native-firebase/firestore": "^12.0.0",
    "react-native": "0.63.4",`

**Comment by mikehardy** [comment]
I would start by looking at how we use the APIs in the documentation and in the E2E tests, e.g.:

https://github.com/invertase/react-native-firebase/tree/master/packages/database/e2e/reference (and everything around there).

**Reply by mikehardy** [reply]
(or since you seem to be mixing "database" and "firestore" (two totally separate modules and storage systems) maybe the firestore e2e tests, or maybe just use one of them, but use it correctly per the API documentation?)

**Reply by AttilaBerczik** [reply]
Could you point me towards a basic example project in which the Realtime Database is used?
Huge thank you for your help! 🙏

**Reply by mikehardy** [reply]
Sure, it's our tests project with e2e tests using the database in every way possible, link as above or https://github.com/invertase/react-native-firebase/blob/master/tests/README.md
